articles/per_package_documentation.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: default
+title: Per-Package Documentation
+permalink: articles/per_package_documentation.html
+abstract: This article describes the requirements and design of ROS 2’s per-package documentation system.
+author: Marya Belanger
+published: true
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Background
+
+ROS 2 is lacking a process for the conglomeration of package documentation.
+Discoverability for package documentation (and documentation in general) is one of the most frequent user complaints.
+
+The goal of this design is to standardize a documentation process for ROS 2 packages to encourage maintainers to robustly document their packages.
+The results will be presented in a consistent way alongside the general ROS 2 documentation, making the documentation as a whole discoverable and consistent.
+
+In general, the vision for the system is:
+
+- Package maintainers document their packages in their repositories following some recommended guidelines or templates
+- Package documentation (API documentation, other package related content) from the repositories is built on the ROS 2 infrastructure and deployed to the ROS 2 documentation site in an automatic process maintainers can opt in to by adding a `doc` block to the `distribution.yaml`
+  - See [REP 141](https://www.ros.org/reps/rep-0141.html) for more context.
+- Package documentation is indexed alongside ROS 2's generic content on the docs site
+
+## Context
+
+The per-package documentation plan is an extension of the general ROS 2 documentation project, which introduced some requirements that affect the context of the per-package documentation:
+
+- All documentation will be hosted on `docs.ros.org`
+- All documentation will be versioned by ROS 2 distribution names
+- The URL structure will be `docs.ros.org/<lang>/<distro>/...` for generic documentation and `docs.ros.org/<lang>/<distro>/p/<package_name>/...` for package documentation
+  - See the diagram below for more context on where documentation will fall under this URL structure:
+  ![](per_package_documentation/Version_package_flowchart.png)
```

Comment:
I'm late to the game, but two things:
 * As a usability requirement, all images should have alt text. 
 * Should independent packages go under `docs.ros.org/<lang>/independent`?

Reply:
@DLu thank you for pointing out the missing alt text! 
And yes, technically everything will fall under `docs.ros.org/<lang>`, we will make that clearer in the next iteration of the document.